docs: describe how to manage embedded plugin registry for VS Code editor (#2503)
eclipse-che/che-docs: modules/administration-guide/partials/proc_adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance.adoc

Completing this procedure results in an optional, custom build of the embedded Open VSX registry for use in workspaces of your organization’s Che.

.Procedure

include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]

** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:
+
[source,json,subs="+quotes"]
----
    {
        "id": "__<published_by>__.__<unique_identifier>__"
    }
----
+
[TIP]
====
The latest extension version on link:https://open-vsx.org[open-vsx.org] is the default. Alternatively, you can add `"version": "__<extension_version>__"` on a new line to specify a version.
====

> ** If it is an extension from link:https://marketplace.visualstudio.com/VSCode[Visual Studio Marketplace] that is not available on link:https://open-vsx.org[Open VSX] yet, you can ask the extension author to publish it on link:https://open-vsx.org[open-vsx.org] according to the link:https://github.com/eclipse/openvsx/wiki/Publishing-Extensions#how-to-publish-an-extension[instructions] so that you can add it as an extension that is published on link:https://open-vsx.org[open-vsx.org].

Reviewer: @nickboldt, yes, exactly. This is what I'm saying too.

:_content-type: PROCEDURE

[id="adding-or-removing-vscode-extensions-in-the-embedded-open-vsx-registry-instance"]
= Adding or removing pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance

You can add or remove your choice of pass:[<!-- vale RedHat.Spelling = NO -->] pass:[<!-- vale RedHat.Spelling = NO -->] VS Code pass:[<!-- vale RedHat.Spelling = YES -->] pass:[<!-- vale RedHat.Spelling = YES -->] extensions in the embedded Open VSX registry instance that is cached by {prod-short} to support air-gapped, offline, and proxy-restricted environments.

This will create a custom build of the Open VSX registry, which can be used in your organization’s workspaces. 

> NOTE: A subscription update reverts your `CheCluster` to default settings, and you must reapply your changes to `CheCluster`. 

Reviewer: @nickboldt, I'd like to tweak your suggestion. Is the following okay?
 
```suggestion
WARNING: Subscription updates for the {prod-short} Operator revert any customizations of the `CheCluster` custom resource to the published settings for the updated operator. After the update is complete, you must reapply your changes to the `CheCluster` custom resource. You may also choose to build a new registry container based on the updated {prod-short} container, in order to get the latest security fixes and updates.
```

.Procedure

include::example$snip_{project-context}-steps-to-add-or-remove-extensions-in-openvsx-sync-json.adoc[]

> ** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including an object with the extension `id` in the format __<published_by>__.__<unique_identifier>__ that you can collect from the extension page on link:https://open-vsx.org[open-vsx.org]:

Reviewer: @nickboldt, I'd like to tweak your suggestion. Is this okay? 
```suggestion
** If the extension is published on link:https://open-vsx.org[open-vsx.org], you can add it by including its extension `id` in the format __<published_by>__.__<unique_identifier>__. You can find these details on the extension's listing page on link:https://open-vsx.org[open-vsx.org].
```
